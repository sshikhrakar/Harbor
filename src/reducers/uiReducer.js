import * as actionTypes from '../actions/actionTypes';

export const initialState = {
  auth: {
    isLoggingIn: false,
    hasLoginErrored: false,
  },
};

/**
 * Holds the UI state for many parts of the app.
 * This reducer key will not be persisted into storage.
 *
 * @param {Object} state=initialState
 * @param {Object} action
 * @returns {Object}
 */
function uiReducer(state = initialState, action) {
  switch(action.type) {

    case actionTypes.LOGIN_VIA_EMAIL:
      return {
        ...state,
        auth: uiAuthReducer(state.auth, action),
      };

    case actionTypes.LOGIN_VIA_EMAIL_FULFILLED:
      return {
        ...state,
        auth: uiAuthReducer(state.auth, action),
      };

    case actionTypes.LOGIN_VIA_EMAIL_ERRORED:
      return {
        ...state,
        auth: uiAuthReducer(state.auth, action),
      };

    default:
      return state;
  }
}

export default uiReducer;

/**
 * Reducer for the ui.Auth state slice.
 *
 * @param {Object} state
 * @param {Object} action
 * @returns {Object}
 */
function uiAuthReducer(state, action) {
  switch(action.type) {
    case actionTypes.LOGIN_VIA_EMAIL:
      return {
        ...state,
        isLoggingIn: true,
      };

    case actionTypes.LOGIN_VIA_EMAIL_FULFILLED:
      return {
        ...state,
        isLoggingIn: false,
        hasLoginErrored: false,
      };

    case actionTypes.LOGIN_VIA_EMAIL_ERRORED:
      return {
        ...state,
        isLoggingIn: false,
        hasLoginErrored: true,
      };
  }
}
