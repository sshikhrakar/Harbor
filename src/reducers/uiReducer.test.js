import uiReducer, { initialState } from './uiReducer';
import {
  loginViaEmail,
  loginViaEmailFulfilled,
  loginViaEmailErrored,
  signupViaEmail,
  signupViaEmailFulfilled,
  signupViaEmailErrored,
  signupViaEmailCancelled,
} from '../actions/authActions';

describe('REDUCER: UI', () => {

  it('should render the initial state when state is undefined and action is an empty object', () => {
    expect(uiReducer(undefined, {})).toEqual(initialState);
  });

  it('should set isLoggingIn to true when loginViaEmail is received', () => {
    const email = 'foobar@gmail.com';
    const password = 'foobar';

    expect(
      uiReducer(undefined, loginViaEmail(email, password))
    ).toMatchSnapshot();
  });


  it('should clear isLoggingIn and error when loginViaEmailFulfilled is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isLoggingIn: Math.random() > 0.5 ? false : true,
          hasLoginErrored: Math.random() > 0.5 ? false : true,
        },
      }, loginViaEmailFulfilled())
    ).toMatchSnapshot();
  });


  it('should set hasLoginErrored and clear isLoggingIn when loginViaEmailErrored is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isLoggingIn: Math.random() > 0.5 ? true : false,
          hasLoginErrored: Math.random() > 0.5 ? true : false,
        },
      }, loginViaEmailErrored())
    ).toMatchSnapshot();
  });


  it('should set isSigningUp to true when signupViaEmail is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isSigningUp: Math.random() > 0.5 ? true : false,
        },
      }, signupViaEmail())
    ).toMatchSnapshot();
  });

  it('should set isSigningUp and hasSignupErrored to false when signupViaEmailFulfilled is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isSigningUp: Math.random() > 0.5 ? true : false,
          hasSignupErrored: Math.random() > 0.5 ? true : false,
        },
      }, signupViaEmailFulfilled())
    ).toEqual({
      ...initialState,
      auth: {
        ...initialState.auth,
        isSigningUp: false,
        hasSignupErrored: false,
      },
    });
  });

  it('should set isSigningUp to false and hasSignupErrored to true when signupViaEmailErrored is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isSigningUp: Math.random() > 0.5 ? true : false,
          hasSignupErrored: Math.random() > 0.5 ? true : false,
        },
      }, signupViaEmailErrored())
    ).toMatchSnapshot();
  });


  it('should set isSigningUp to false and hasSignupErrored to true when signupViaEmailErrored is received', () => {
    expect(
      uiReducer({
        ...initialState,
        auth: {
          ...initialState.auth,
          isSigningUp: Math.random() > 0.5 ? true : false,
          hasSignupErrored: Math.random() > 0.5 ? true : false,
        },
      }, signupViaEmailCancelled())
    ).toMatchSnapshot();
  });


});
