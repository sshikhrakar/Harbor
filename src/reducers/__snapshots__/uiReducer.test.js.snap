// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCER: UI should clear isLoggingIn and error when loginViaEmailFulfilled is received 1`] = `
Object {
  "auth": Object {
    "hasLoginErrored": false,
    "hasSignupErrored": false,
    "isLoggingIn": false,
    "isSigningUp": false,
  },
}
`;

exports[`REDUCER: UI should set hasLoginErrored and clear isLoggingIn when loginViaEmailErrored is received 1`] = `
Object {
  "auth": Object {
    "hasLoginErrored": true,
    "hasSignupErrored": false,
    "isLoggingIn": false,
    "isSigningUp": false,
  },
}
`;

exports[`REDUCER: UI should set isLoggingIn to true when loginViaEmail is received 1`] = `
Object {
  "auth": Object {
    "hasLoginErrored": false,
    "hasSignupErrored": false,
    "isLoggingIn": true,
    "isSigningUp": false,
  },
}
`;

exports[`REDUCER: UI should set isSigningUp to false and hasSignupErrored to true when signupViaEmailErrored is received 1`] = `
Object {
  "auth": Object {
    "hasLoginErrored": false,
    "hasSignupErrored": true,
    "isLoggingIn": false,
    "isSigningUp": false,
  },
}
`;

exports[`REDUCER: UI should set isSigningUp to false and hasSignupErrored to true when signupViaEmailErrored is received 2`] = `
Object {
  "auth": Object {
    "hasLoginErrored": false,
    "hasSignupErrored": true,
    "isLoggingIn": false,
    "isSigningUp": true,
  },
}
`;

exports[`REDUCER: UI should set isSigningUp to true when signupViaEmail is received 1`] = `
Object {
  "auth": Object {
    "hasLoginErrored": false,
    "hasSignupErrored": false,
    "isLoggingIn": false,
    "isSigningUp": true,
  },
}
`;
